src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
 
     @Override
     public void put(K key, V value) {
+        if (table == null) {
+            resize();
+        }
 
+        int indexOfBucket = hash(key);
+
+        if (table[indexOfBucket] == null) {
+            table[indexOfBucket] = new Node(indexOfBucket, key, value, null);
+            size++;
+        }
+
+        Node<K, V> pointer = table[indexOfBucket];
+
+        while (pointer != null) {
+            if (pointer.key == key || key != null && key.equals(pointer.key)) {
+                pointer.value = value;
+                break;
+            }
+            if (pointer.next == null) {
+                pointer.next = new Node(indexOfBucket, key, value, null);
+                size++;
+                break;
+            } else {
+                pointer = pointer.next;
+            }
+        }
```

Comment:
try to not overcomlicate `if`s, loops, and logic inside them
```suggestion
            if (pointer.key == key || key != null && key.equals(pointer.key)) {
                pointer.value = value;
                return;
            }
            pointer = pointer.next;
        }
        pointer = new Node(key, value);
```